src/plugins/data_source_management/public/components/create_data_source_wizard/components/create_form/create_data_source_form.tsx

```diff
@@ -583,24 +595,33 @@ export class CreateDataSourceForm extends React.Component<
               <EuiText>
                 <FormattedMessage
                   id="dataSourcesManagement.createDataSource.authenticationMethodDescription"
-                  defaultMessage="Enter the authentication details to access the endpoint. If no authentication is required, select "
+                  defaultMessage="Enter the authentication details to access the endpoint."
                 />
-                <b>
+                {this.authOptions.length > 0 && (
                   <FormattedMessage
-                    id="dataSourcesManagement.createDataSource.noAuthentication"
-                    defaultMessage="No authentication"
+                    id="dataSourcesManagement.createDataSource.authenticationMethodDescription"
+                    defaultMessage=" If no authentication is required, select "
                   />
-                </b>
+                )}
+                {this.authOptions.length > 0 && (
+                  <b>
+                    <FormattedMessage
+                      id="dataSourcesManagement.createDataSource.noAuthentication"
+                      defaultMessage="No authentication"
+                    />
+                  </b>
+                )}
```

Comment:
I don't understand this piece of code. Why we are checking length of authOptions?

Reply:
Thanks for the comment and good catch.

Will updated this condition to check if NoAuth supported or not.

Reply:
Condition updated. Thanks~